### PR TITLE
Refactor MLine Media to follow qucs implementation

### PR DESCRIPTION
**Pull request by mhuser**
Refactor MLine Media against [qucsator implementation](https://github.com/Qucs/qucsator/blob/develop/src/components/microstrip/msline.cpp) for reference, add test cases and modify some notebooks if required .

### CODE REVIEW

File: skrf/media/mline.py [hunk 1]
```diff
@@ -89,7 +90,7 @@ class MLine(Media):
         * 'hammerstadjensen'
         * 'yamashita'
         * 'kobayashi'
-        * None
+        * 'none'
```
Comment (jhillairet) [inline skrf/media/mline.py:93]: could you add in the docstring the default values? Ex:

```
* 'kirschningjansen' (default)
* 'hammerstadjensen'
```

**Review by jhillairet — APPROVED** [review]
OK. Do you plan to add additional tests?

**Comment by mhuser** [comment]
> OK. Do you plan to add additional tests?

Yes. It is still a work on progress. I still need to:

- [ ] Add a test for each model against qucs generated s-parameters
- [ ] Add a test to ensure warning is raised when t < 3*skin_depth 
- [ ] Repair some notebooks that use MLine
- [ ] Think more about this media attributes and docstrings e.g. if it provide `alpha_conductor` with `return self._alpha_conductor`

**Comment by mhuser** [comment]
Of course I am open to more tests ideas !

**Comment by mhuser** [comment]
Almost ready.
There is still the question if I add Wheeler and Schneider impedance models -not that much works to port these from qucs -
and also some more works to compare with the ads s-parameters provided by @JAnderson419 (may he be thanked a lot here)!

**Review by jhillairet — CHANGES_REQUESTED** [review]
just a comment concerning the name of the functions. Otherwise OK to me, thank you.

File: skrf/media/mline.py [hunk 2]
```diff
@@ -140,258 +161,254 @@ def __str__(self) -> str:
...
 
-    @property
-    def ep_r_f(self) -> NumberLike:
+        Returns
+        -------
+        Z0 : :class:`numpy.ndarray`
+        """
+        return self._z_characteristic
+    
+    def analyseDielectric(self, ep_r: NumberLike, tand: NumberLike,
```
[the remark this answers is not in the document]
Comment (mhuser) [reply skrf/media/mline.py]: Indeed. Done.

File: skrf/media/tests/test_mline.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        self.assertTrue(npy.all(npy.abs(l2.s_db - self.ref2.s_db) < 0.1))
+        self.assertTrue(npy.all(npy.abs(l2.s_deg - self.ref2.s_deg) < 1.))
+        self.assertTrue(npy.all(npy.abs(l3.s_db - self.ref3.s_db) < 0.1))
+        self.assertTrue(npy.all(npy.abs(l3.s_deg - self.ref3.s_deg) < 1.))
+        self.assertTrue(npy.all(npy.abs(l4.s_db - self.ref4.s_db) < 0.1))
+        self.assertTrue(npy.all(npy.abs(l4.s_deg - self.ref4.s_deg) < 1.))
+        # fixme: cannot pass currently
+        #self.assertTrue(npy.all(npy.abs(l4.s_db - self.ref4.s_db) < 0.1))
+        #self.assertTrue(npy.all(npy.abs(l5.s_deg - self.ref5.s_deg) < 1.))
+
```
[the remark this answers is not in the document]
Comment (mhuser) [reply skrf/media/tests/test_mline.py:128]: I am on that. Seems my implementation Djordjevic/Svensson dielectric frequency variation is not consistent with ADS one. This kind of dispersion does not exists in Qucs and was done from the paper.

[the remark this answers is not in the document]
Comment (mhuser) [reply skrf/media/tests/test_mline.py:128]: I am returning this PR to draft until either I understand or I write in the docstring that this functionnality is not well tested and lead to different results that other simulators.

**Comment by mhuser** [comment]
Current state of consistency test between `mline` and with Djordjevic/Svensson dielectric frequency dispersion ADS.
Still some works left.
image

**Comment by mhuser** [comment]
Agreement with ADS significantly improved.
Only a refinement of docstring still required. I will mention the papers reference for each methods.
image

File: skrf/media/tests/test_mline.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mhuser) [reply skrf/media/tests/test_mline.py]: The tests now pass and agree with ADS data. The limit for S11 at low-frequency has still to be 10x more than 0.1 dB and 1°, but this looks not bad at all and a lot better as before.

**Comment by Ttl** [comment]
It looks like the second to last revert commit reverted also all the previous commits. There isn't currently any difference to the master branch.

I don't like the `compatibility_mode` parameter very much. Only one of the settings should be correct and I don't think we should try to keep compatibility with other simulator models. I don't think it's a parameter that many users would change anyway.

**Comment by mhuser** [comment]
Looks like I created a git mess.

**Comment by mhuser** [comment]
History now looks ugly, but the changes should be all back.

**Comment by mhuser** [comment]
> I don't like the `compatibility_mode` parameter very much. Only one of the settings should be correct and I don't think we should try to keep compatibility with other simulator models. I don't think it's a parameter that many users would change anyway.

I understand your point of view, and this parameter make the code be less clear. It was more a trick to ease the comparison between both set of results. However, if we keep only one or the other behaviour, then we cannot test all functionalities against a known reference (some being implemented only in ADS or only in QUCS).

It could also be tough as an "almost hidden" feature, that most users would never change or be aware of. The question being which behaviour to pick as the default. From the elements you presented, the model closer to RGLC would be the ADS one, which will require some minor changes to notebooks before this PR could be merged.

Have you other ideas on how to handle this ?

**Review by Ttl — APPROVED** [review]

**Review by jhillairet — APPROVED** [review]
Great work, thank you!

**Comment by jhillairet** [comment]
If you are OK to merge, let us know.

**Comment by mhuser** [comment]
OK

**Comment by mhuser** [comment]
Thank you so much !

**Pull request merged**